[ADD] Module github_product_creator

espo-tony opened the pull request.

espo-tony committed:
[10.0][ADD] Mdoule github_product_creator

espo-tony committed:
[10.0][ADD] Mdoule github_product_creator

espo-tony committed:
[FIX] Moved functionality to link module version to product variant from product template creation to product variant creation

espo-tony committed:
[FIX] Moved functionality to link module version to product variant from product template creation to product variant creation

hurrinico commented on github_product_creator/data/product_attribute.xml:
```diff
@@ -0,0 +1,28 @@
+        <field name="name">10.0</field>
+        <field name="attribute_id" ref="attribute_odoo_version"/>
+    </record>
+
```
could be an idea to already add 11.0?

tafaRU committed:
[IMP] Apply some flake8 clean up

tafaRU committed:
[IMP] Apply some flake8 clean up

tafaRU committed:
[IMP] Improve README

tafaRU committed:
[IMP] Improve README

tafaRU committed:
Rename module to apps_product_creator

tafaRU committed:
Rename module to apps_product_creator

tafaRU committed:
[IMP] Replace openerp imports to odoo

tafaRU committed:
[IMP] Replace openerp imports to odoo

espo-tony committed:
Merge pull request #1 from tafaRU/10.0-github_product_creator-imp

10.0 GitHub product creator imp

tafaRU committed:
[ADD] apps_product_creator: add cron_create_product

tafaRU committed:
[ADD] apps_product_creator: add cron_create_product

tafaRU committed:
[ADD] apps_product_creator: add cron_create_product

espo-tony committed:
Merge pull request #2 from tafaRU/10.0-github_product_creator-imp

[ADD] apps_product_creator: add cron_create_product

RoelAdriaans commented on apps_product_creator/README.rst:
```diff
@@ -0,0 +1,92 @@
+
+#. Go to ...
+
+.. figure:: path/to/local/image.png
```
We need to create and find an image

RoelAdriaans commented on apps_product_creator/README.rst:
```diff
@@ -0,0 +1,92 @@
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
```
Add the repo id, nummber 249 , branch 10.0

RoelAdriaans commented on apps_product_creator/README.rst:
```diff
@@ -0,0 +1,92 @@
+Do not contact contributors directly about support or help with technical issues.
+
+Funders
+-------
```
This can probably be removed, we don't have funders

RoelAdriaans reviewed (changes requested):
We need to add the grap repository for the gitlab interface to the oca_dependencies.txt file, to add the dependencies for runbot and the travis build.

espo-tony committed:
[FIX] Changed field name for compatibility with base code

espo-tony committed:
[FIX] Changed field name for compatibility with base code

espo-tony committed:
[FIX] Refactoring of product.product form view definition

espo-tony committed:
[FIX] Refactoring of product.product form view definition

espo-tony committed:
[FIX] Implemented module image handling

espo-tony committed:
[FIX] Implemented module image handling

acsonefho committed:
Improvements: Edit readme + improvement about code

acsonefho committed:
FIX: fix conflicts and remove depracated api.one

espo-tony committed:
Merge pull request #5 from tafaRU/10.0-apps_product_creator

10.0 apps product creator

espo-tony committed:
[FIX] Changed product form view definition

acsonefho committed:
FIX: licence

acsonefho committed:
FIX: fix readme

espo-tony committed:
Merge pull request #6 from acsonefho/10_apps_product_creator

10 apps product creator

Commit:
[FIX] fix attr_xml_id and domain on cron search

espo-tony committed:
Merge pull request #7 from hurrinico/xml-id-error_and-cron

[FIX] fix attr_xml_id and domain on cron search

hurrinico reviewed (approved):
once flake8 is fixed :+1: 
code review and functional test

elicoidal commented on apps_product_creator/README.rst:
```diff
@@ -0,0 +1,60 @@
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/249/10.0
+
+Known issues / Roadmap
```
Remove if unused

elicoidal commented on apps_product_creator/README.rst:
```diff
@@ -0,0 +1,60 @@
+* Antonio Esposito <a.esposito@onestein.nl>
+* Nicola Malcontenti <nicola.malcontenti@agilebg.com>
+* Alex Comba <alex.comba@agilebg.com>
+
```
Please add my name: `* Eric Caudal <eric.caudal@elico-corp.com>`

elicoidal commented on apps_product_creator/README.rst:
```diff
@@ -0,0 +1,60 @@
+* Nicola Malcontenti <nicola.malcontenti@agilebg.com>
+* Alex Comba <alex.comba@agilebg.com>
+
+Do not contact contributors directly about support or help with technical issues.
```
Remove this line

elicoidal commented on apps_product_creator/data/ir_cron.xml:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--# Copyright (C) 2016-Today: Odoo Community Association (OCA)-->
```
no need the "#"

elicoidal reviewed (changes requested):
Almost there. not tested yet. 
When the module is stable README to be popuolated

elicoidal commented:
CLA should be OK (both CLA of Acsone and Onestein)

espo-tony committed:
[FIX] Some small refactoring

astirpe commented on apps_product_creator/README.rst:
```diff
@@ -0,0 +1,53 @@
+Apps Product creator
+====================
+
+This module extends the functionality of the module Github connector to create a product for every module found.
```
The description does not mention anything regarding the cron job that is provided within this module. I think that this description should clarify how the mechanism works. Adding a short Configuration/Usage description would be nicer.

elicoidal replied:
screenshots are welcome (when the product is stable)

astirpe commented on apps_product_creator/data/ir_cron.xml:
```diff
@@ -0,0 +1,19 @@
+<!--License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).-->
+<odoo noupdate="1">
+
+        <record model="ir.cron" id="cron_update_organization">
```
When declaring a record, usually the `id` is placed [before the rest](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#format). Check also the other `.xml` files.

astirpe commented on apps_product_creator/models/product_product.py:
```diff
@@ -0,0 +1,214 @@
+        'odoo.module.version',
+        'Odoo Module',
+    )
+    license = fields.Char(
```
I'm thinking about the definition of fields like license, summary, version, author, etc...
Such fields have a very generic technical name, and they could conflict with some other module that extends and adds fields to `product.product`.

I understand that their related fields have the same name, but the model `product.product` is more generic and widely extended, compared to the other model `odoo.module`.

Maybe they could renamed by adding a prefix, like `apps_` for example, in order to make them more related to this particular module.

elicoidal replied:
Why not o2m fields as the license is defined in interface-github module

astirpe commented on apps_product_creator/views/view_odoo_module.xml:
```diff
@@ -0,0 +1,15 @@
+        <field name="inherit_id" ref="github_connector_odoo.view_odoo_module_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//div[@name='button_box']" position="inside">
+                <button name="action_create_product" string="Create Products" type="object" class="oe_highlight" />
```
For testing purposes the Create Products button is fine. But in a production scenario I would instead expect a smart button to view products.

Something like this:


```
    <record id="action_odoo_module_product" model="ir.actions.act_window">
        <field name="name">Odoo Module Product</field>
        <field name="type">ir.actions.act_window</field>
        <field name="res_model">product.product</field>
        <field name="view_type">form</field>
        <field name="view_mode">kanban,tree,graph,form</field>
    </record>

    <record id="odoo_module_form_view" model="ir.ui.view">
        <field name="model">odoo.module</field>
        <field name="inherit_id" ref="github_connector_odoo.view_odoo_module_form"/>
        <field name="arch" type="xml">
            <xpath expr="//div[@name='button_box']" position="inside">
                <button name="%(action_odoo_module_product)d" type="action" class="oe_stat_button" icon="fa-cube"
                        attrs="{'invisible': [('module_version_qty', '=', False)]}"
                        context="{'search_default_module_id': active_id}">
                    <field name="module_version_qty" string="Products" widget="statinfo" />
                </button>
                <button name="action_create_product" string="Create Products" type="object"
                        class="btn btn-default" icon="fa-chevron-right"
                        groups="github_connector.group_github_connector_manager" />
            </xpath>
        </field>
    </record>
```

Will the Create Products button be removed in a final version of this module?

astirpe commented on apps_product_creator/models/odoo_module.py:
```diff
@@ -0,0 +1,97 @@
+                lambda p: p.odoo_module_id == odoo_module)
+            if not product:
+                product_values = odoo_module._get_template_values()
+                new_product = product_obj.create(product_values)
```
I'm doing some tests, creating products from OCA modules. All the products are created with Sale Price set to 1euro. I was expecting a Sale Price = 0 instead.

Also the flag Can be Purchased is set to True, I was expecting to be False.

elicoidal replied:
for Sales price, versions ago, SP = 0 was bringing errors. Not sure this is the case today
Product should be:
* not purchasable
* Saleable
* ~service (digital)~ (edited: https://github.com/OCA/apps-store/pull/2/files#r143997887)
* Cost = 0 / SP = 0 (if possible)

astirpe commented on apps_product_creator/README.rst:
```diff
@@ -0,0 +1,53 @@
+
+This module extends the functionality of the module Github connector to create a product for every module found.
+A variant is also created for every version of the module.
+
```
To make some tests on a fresh database I had to enable product variants. Maybe adding a remind to enable product variants, for example adding a line in the Usage section, would be an help.

elicoidal replied:
indeed but better in the configuration section.
Same would apply for digital products: shall the module depend on website_sale_digital?

espo-tony replied:
Do digital products need to be marked as such in some particular way? For what I know, module website_sale_digital doesn't add fields or options to a product, rather allows its attachments to be downloaded from the website interface.
If such is the case, I don't think this module should directly depend on it. What do you think?

astirpe replied:
The other module _apps_download_ https://github.com/OCA/apps-store/pull/3 is the one that depends on _website_sale_digital_

elicoidal replied:
Crystal clear!

astirpe commented on apps_product_creator/views/view_product_product.xml:
```diff
@@ -0,0 +1,33 @@
+    <record id="product_product_form_view" model="ir.ui.view">
+        <field name="name">product.product.form (in github_product_creator)</field>
+        <field name="model">product.product</field>
+        <field name="priority" eval="100"/>
```
Why `<field name="priority" eval="100"/>` is here? I'm trying to put low values and it seems working fine as well...

astirpe commented on apps_product_creator/views/view_product_product.xml:
```diff
@@ -0,0 +1,33 @@
+<odoo>
+
+    <record id="product_product_form_view" model="ir.ui.view">
+        <field name="name">product.product.form (in github_product_creator)</field>
```
`<field name="name" ...` can be removed here. The ORM generates automatically a name for the view.

astirpe commented on apps_product_creator/views/view_odoo_module.xml:
```diff
@@ -0,0 +1,15 @@
+<!--License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).-->
+<odoo>
+    <record id="odoo_module_form_view" model="ir.ui.view">
+        <field name="name">odoo.module.form (in github_product_creator)</field>
```
`<field name="name" ...` can be removed here. The ORM generates automatically a name for the view.

astirpe commented on apps_product_creator/views/view_product_product.xml:
```diff
@@ -0,0 +1,33 @@
+        <field name="inherit_id" ref="product.product_normal_form_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="//notebook" position="inside">
+                <page string="Module Informations" id="module_informations" attrs="{'invisible':[('odoo_module_version_id','=',False)]}">
```
Replace `id=` with `name=`.

Also I'm noticing that the tab "Module Informations" is missing for `product.template`: it is only present for `product.product`. Is it made on purpose?

astirpe commented on apps_product_creator/models/product_template.py:
```diff
@@ -0,0 +1,25 @@
+        'Icon Image',
+        readonly=True,
+        related="odoo_module_id.image",
+        store=True,
```
Here the image_module is a Binary and also a related field, why is it set as stored?
Since it is a binary, I would expect nobody will need to search/sort this field... In my opinion there's not need to store it.

astirpe commented on apps_product_creator/models/product_product.py:
```diff
@@ -0,0 +1,214 @@
+    image = fields.Binary(
+        "Big-sized image",
+        compute='_compute_images',
+        inverse='_set_image',
```
Fix this warning:

```
apps_product_creator/models/product_product.py:72: [C8110(method-inverse), ProductProduct] Name of inverse method should start with "_inverse_"
apps_product_creator/models/product_product.py:80: [C8110(method-inverse), ProductProduct] Name of inverse method should start with "_inverse_"
apps_product_creator/models/product_product.py:87: [C8110(method-inverse), ProductProduct] Name of inverse method should start with "_inverse_"

astirpe commented on apps_product_creator/models/odoo_module.py:
```diff
@@ -0,0 +1,97 @@
+class OdooModule(models.Model):
+    _inherit = 'odoo.module'
+
+    product_template_id = fields.Many2one(
```
The relation between _odoo.module_ and _product.template_ is a `'0..1 to 0..1'`, in the sense that one _odoo.module_ can have at most one _product.template_, and the same for the other way round. So here defining a  Many2one `product_template_id` is fine.

But  I'm thinking that converting the field `product_template_id` to a _One2many_ `product_template_ids` would result more clear and it also simplifies the code all around. For example in the `def _create_product()` you will not need to filter the matching product inside a loop, because you can read `product_template_ids` directly.

What do you think?

elicoidal replied:
I am not technically of the advantage but it looks confusing to me @astirpe

astirpe reviewed (changes requested):
I tested this module on a fresh db, and it works fine.

The unit tests are still missing.

To let Travis and Runbot go ahead, you can temporarily change the `oca_dependencies.txt` file.
In your branch, replace the content of `oca_dependencies.txt` with the following:

```
interface-github https://github.com/grap/interface-github 10.0_ADD_github_connector_github_connector_odoo
```

Once the `github_connector*` modules will be merged, restore the `oca_dependencies.txt` file.

hurrinico commented:
i agree with the @astirpe code review.
Functionally works fine!

elicoidal commented on apps_product_creator/__manifest__.py:
```diff
@@ -0,0 +1,22 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Github product creator',
+    'summary': 'Create a product variant (by version) for every module found.',
```
out of context, this description is cryptic...
I would suggest something like:
"Create a product variant (by version) for every module found in the Github repositories"

elicoidal commented on apps_product_creator/data/product_attribute.xml:
```diff
@@ -0,0 +1,25 @@
+        <field name="attribute_id" ref="attribute_odoo_version"/>
+    </record>
+
+    <record model="product.attribute.value" id="odoo_version_100">
```
Why not add all possible versions:
5, 6, 6.1, 7 and 11? (at least 7 and 11)

elicoidal commented on apps_product_creator/models/product_product.py:
```diff
@@ -0,0 +1,214 @@
+        store=True,
+    )
+    description_rst_html = fields.Html(
+        'HTML the RST Description',
```
HTML of the...

elicoidal commented on apps_product_creator/models/product_product.py:
```diff
@@ -0,0 +1,214 @@
+        related="odoo_module_version_id.description_rst_html",
+        store=True,
+    )
+    version = fields.Char(
```
I would expect a o2m field (all the more if we define it in the data)

elicoidal commented on apps_product_creator/models/product_product.py:
```diff
@@ -0,0 +1,214 @@
+        :param attribute: product.attribute.value recordset
+        :return: odoo.module.version recordset
+        """
+        # Attribute value (so the name field) must be the milestone name
```
I am not comfortable using milestones. Our repos and versions are based on branches, not milestones (barely used)

espo-tony replied:
In this module we use the milestone related to a branch, simply as a shortcut to get the version of that branch.
This choice is also due to the technical constraint imposed by the module github_connector_odoo which uses the model github.organization.milestone for keeping track of the versions available for any module.

I'm not perfectly happy with this choice neither but at the moment I don't see a better solution. You have suggestions on how to get the branch version on a branch, based on data provided by the github-connector modules?

elicoidal replied:
Then we need to bring the topic to @legalsylvain before it is too late.
Basically we should be able to create versions of the module based on the branch it is located.

legalsylvain replied:
I don't understand exactly the problem. Is it just a name problem ? or it is a feature problem ? 
milestone are just some branches. for exemple (l10n_brazil branches ) : 
![image](https://user-images.githubusercontent.com/3407482/31510625-547a0d74-af85-11e7-82c5-59e91c3296d8.png)

6.0 / 6.1 / 7.0 / 8.0 / 9.0 / 10.0 / 11.0 are branches and milestones (I mean, a stable and usable revision)
gh-pages is a branch, but not a milestone.

In a project with many repositories, like OCA, milestone are defined by the organization. (7, 8, 9...) for that reason, i added the model github.organization.milestone. 

regards.

sbidoul replied:
Hi all, 

Please do not rely on the milestone to determine the odoo series for which an addon applies. Use the branch prefix (the part before the first dash), or the version in the manifest (the first two digits, reliable from version 8 onwards).

We must leave the liberty to maintainers to use milestones for what they are made for, ie organizing and planning future releases.

legalsylvain replied:
Again, I'm not using github milestone. (no synchronisation). I'm using "some" branches. That is the objective of github.organization.milestone. (to define which branches are relevant). 

the previous name of this model was github.organization.serie and community asked to change the name of the model and milestone was OK. 

https://github.com/OCA/interface-github/pull/2/commits/a96d476e1b6d38f59aee9909ec29ed3bd4b50d79#diff-ae93b445bcca140d9d99caa33763c37f

sbidoul replied:
Thanks for the clarification @legalsylvain, I was confused by the use of the milestone name. Series was much more clear to me, sorry I missed that discussion.

legalsylvain replied:
> Series was much more clear to me,

Me too ! ;-)

The discussion is here.
https://github.com/OCA/interface-github/pull/1#discussion_r93159698

@lasley, @elicoidal  : could you give your feedback, and choose a name that is OK for all ? 

regards.

elicoidal replied:
@legalsylvain @sbidoul 
Originally, series is a notion which does not exist in github so I suggested to stick to github naming.
Milestones are currently not useful in our organization because they are not consistently used.
We need to stick to the branch name or as suggested by @sbidoul the module manifest.

I would suggest interface-github module to highlight the branch the module is in (whatever the milestone) and apps-store module to use the assumption that one branch is one version (with a crappy text compare to the internal version model to rule out WIP branches).
I am open to improve this: suggestions welcome!

sbidoul replied:
@elicoidal *organization level* milestone is not a github concept either AFAICT. We are trying to model an odoo specific concept here.

elicoidal replied:
@sbidoul then maybe I am confused with the naming here. I thought about Milestones as defined here after:
![screenshot from 2017-10-13 11-59-29](https://user-images.githubusercontent.com/7600613/31541241-4d047992-b00e-11e7-886f-b8dc006e7e4d.png)

sbidoul replied:
@elicoidal these are project level milestones and not synchronized by github_connector. See also https://github.com/OCA/interface-github/pull/1#discussion_r144515098.

elicoidal replied:
@sbidoul OK: I have never seen other milestones in the interface for the organization.
As said: any name is good to me as long it is consistent.
As for using them for the module version, we would agree that this is probably not the best option

elicoidal replied:
please note: https://github.com/OCA/interface-github/pull/2#issuecomment-336509906

espo-tony committed:
[FIX] Big refactoring
- Renamed attribute names on product.product model;
- Removed some unused attribute definition on product.product model;
- Attribute image_module on product.template is now not stored;
- Views definition made compliant with OCA guidelines;
- Added values 5, 6, 6.1, 7 and 11 to product attribute Odoo Version;
- On odoo.module form views, replaced "Create Products" with Smartbutton for showing related product variants;
 - Products created from modules are now set as not purchasable and with 0 cost and sell price;
 - Rearranged product.product view definition in "Module Informations" tab;

espo-tony commented:
@astirpe @elicoidal 
In my last commit I addressed most of the issues you raised in your reviews. May I ask you to review it again?

About the description (eventually inclusive of screenshots), I'll update it once the module is stable enough, as configurations and usage could still change until then.

elicoidal commented on apps_product_creator/__manifest__.py:
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016-Today: Odoo Community Association (OCA)
```
technically 2017-today

elicoidal commented on apps_product_creator/models/__init__.py:
```diff
@@ -0,0 +1,4 @@
+# -#- coding: utf-8 -#-
```
looks like too many `#`

elicoidal reviewed (changes requested):
Most important is the design decision to be taken to synchronize the version of the module
cc @legalsylvain

espo-tony committed:
[FIX] Adapted to recent changes in dependency module github_connector

espo-tony committed:
[FIX] Fixed Travis

elicoidal commented:
@espo-tony Travis seems unhappy.
Besides that I think we are almost there right?

astirpe commented:
@espo-tony Travis can be temporarily fixed this way: https://github.com/OCA/apps-store/pull/2#pullrequestreview-68276926
Will you create some unit tests?

elicoidal commented:
> Will you create some unit tests?

:+1:

elicoidal commented:
@espo-tony https://github.com/OCA/interface-github/pull/2 is now merged.
Can you update us on the status of the PR?

espo-tony commented:
@elicoidal, @astirpe 
Unfortunately I couldn't find time for working on this lately. I plan to work on tests and Travis somewhere next week.

lasley reviewed (commented):
Thanks @espo-tony. Do you plan on adding tests?

espo-tony committed:
Implemented Unit tests

espo-tony committed:
Implemented Unit tests

espo-tony commented:
@elicoidal , @astirpe , @lasley 
Unit tests added and Travis all green. Does any of you know why the runbot doesn't run?

lasley commented:
@espo-tony - this branch does not have a Travis file. Travis is working because it runs against the proposed merge, but Runbot runs against your branch alone. Rebase on OCA/11.0 will solve

elicoidal commented:
@lasley you mean OCA/10.0 right?

espo-tony committed:
Merge remote-tracking branch 'origin/10_add_github_product_creator' into 10_add_github_product_creator

# Conflicts:
#	apps_product_creator/README.rst
#	apps_product_creator/__init__.py
#	apps_product_creator/__manifest__.py
#	apps_product_creator/data/ir_cron.xml
#	apps_product_creator/data/product_attribute.xml
#	apps_product_creator/models/__init__.py
#	apps_product_creator/models/odoo_module.py
#	apps_product_creator/models/product_product.py
#	apps_product_creator/models/product_template.py
#	apps_product_creator/views/view_odoo_module.xml
#	apps_product_creator/views/view_product_product.xml

astirpe reviewed (approved):
Time to merge IMO :)

espo-tony commented:
@lasley Thanks!

elicoidal commented:
@RoelAdriaans-B-informed can you review again?

elicoidal commented:
cc @sbidoul @lasley @legalsylvain

legalsylvain commented on apps_product_creator/data/product_attribute.xml:
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--Copyright (C) 2017-Today: Odoo Community Association (OCA)-->
```
I think this data should be demo and not data. This is custom implementation. 

like : https://github.com/OCA/interface-github/blob/10.0/github_connector/demo/github_organization_serie.xml

legalsylvain commented on apps_product_creator/models/odoo_module.py:
```diff
@@ -0,0 +1,131 @@
+
+    @api.depends(
+        'product_template_id',
+        'product_template_id.product_variant_ids'
```
Just a question : if product_template_id.product_variant_ids is set, is 'product_template_id' necessary ?

espo-tony replied:
As far as I know, it should be necessary because with only _product_template_id.product_variant_ids_, the method would be triggered only if something changes on the _product.product_ records related on _product_template_id.product_variant_ids_, not if _product_template_id_ itself is edited.

legalsylvain replied:
OK. Thanks for the precision. I didn't know.

legalsylvain commented on apps_product_creator/models/odoo_module.py:
```diff
@@ -0,0 +1,131 @@
+        'product.template',
+        "Product Template",
+    )
+    rel_product_count = fields.Integer(
```
product_qty is maybe more understandable. (rel is not very used in the name of fields).

legalsylvain commented on apps_product_creator/models/odoo_module.py:
```diff
@@ -0,0 +1,131 @@
+        return products
+
+    @api.multi
+    def _get_template_values(self):
```
The name of such function is generally _prepare_template()

legalsylvain commented on apps_product_creator/models/odoo_module.py:
```diff
@@ -0,0 +1,131 @@
+
+    @api.multi
+    def write(self, values):
+        """
```
I think that comment on write (and above on create) functions are useless, as it is a standard in Odoo.

legalsylvain commented on apps_product_creator/models/product_product.py:
```diff
@@ -0,0 +1,183 @@
+    def create(self, values):
+        """
+        Apply the behaviour to product used to save modules.
+        :param values: dict
```
same here.

legalsylvain commented on apps_product_creator/models/odoo_module.py:
```diff
@@ -0,0 +1,131 @@
+    )
+    rel_product_count = fields.Integer(
+        '# of Products',
+        compute='_get_rel_product_count',
```
The name is generally ```_compute_field_name``` in OCA convention.

legalsylvain reviewed (changes requested):
Hi. 
Thanks for this contribution ! Some optional comments inline. (except demo data comment I think it's important to change).
kind regards.

(code review / no test).

espo-tony committed:
[FIX] Refactoring

espo-tony committed:
[FIX] Bugfixing

elicoidal commented:
I think this one is ready to roll.
@RoelAdriaans-B-informed @legalsylvain could you review again?
Thanks

legalsylvain commented on apps_product_creator/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+        'product',
+    ],
+    'data': [
+        'demo/product_attribute.xml',
```
;-) well, could you move this item in the 'demo' key ?

legalsylvain commented:
LGTM except the demo file that is not in the 'demo' key in the __openerp.py__ file.
regards.

elicoidal commented:
Thanks @legalsylvain

espo-tony committed:
[FIX] Moved demo file in demo key within manifest

espo-tony commented:
My bad. It should be in the right position, now.

astirpe commented:
@RoelAdriaans-B-informed your approval is still pending, will you make the final review?

astirpe commented:
@elicoidal this one can be merged IMO. The label "Work in progress" can be removed anyway.

bizzappdev commented:
@elicoidal @espo-tony 
IMO we should add some clarification to identify that if the product/Template is for module or something else? Because on [odoo-community.org](https://odoo-community.org/shop) I can see we have already some products which are not related to Modules.
on second thought we can also use `odoo_module_id` field to identification.

elicoidal commented:
> IMO we should add some clarification to identify that if the product/Template is for module or something else?

For me if odoo_module_id is not empty, it is an easy way to clarify the product status. I don't think we need a function is_module_product()  for that but if there is a technical reason for it, why not
@bizzappdev

elicoidal commented:
I think we can merge this one :smile:

elicoidal commented:
@espo-tony I dont know why Travis is reporting this PR as red :cry: 
https://travis-ci.org/OCA/apps-store
(something about Transifex: I am not sure anymore how to get the right password)

pedrobaeza commented:
@elicoidal, use this secret
```yaml
  secure: "3ZwXX7d/5/L//SDHXQtqiXcjveCb8Y01OyWzfhmLD5tUADmQ4OaZsTQgFw9Z3ONfTBUBfx5fhwe/paxMz1NzzOs3xudQqYDjuT/bTv81vZOC8LoV3q3Rjp5Ogs5qk1TW6B8KzSTrNtEw4od8Ruok9cjqW9sdaew3Fy2jkNg/vStPy+rWVuTAVzi86LedCQbi/remDxNqbSuUY7XAHXG0jmPpqsKFlxa4lu4na6DClJhL3HwPKCnVn4ME4DHsOvqcxcm58oEXXtr2r8ME+WilDSBElK07GRkqLh5AJ3ELgwWzm8ZKX51tieUTMiWa2rqx5VZRMsRnNVcDFlgPu4aLlPcNEK7HZJ6Zd+0yj2OX/rje2Cr9ddZWO77xSZTNwu7lvE950ZuNuAj16jfl2+JyZIlNootFZ/e7VarSP8QnTjt6Y+L+ywdlJih87REFMOxvylPbcpXEBu01wsM5IhxN0PDhzB/a+IsFkqsx74pUg9em10koyi2KBW8CP/N/hc1Yqs/oCEuMDZl3nR6GfaxTEohQT4/5EGeiW5yDZ3iqAu4PiPlUcat05FjoByydF8uSvlYaBf+VKrd8jZLf/b8ALXeXD9s9ne7KWmFvgLe7JjKPk9wdlqGAgW/YoIVIZsbkFbABcykmOGioUTXDbEPdhQKOk8Wzw1baYIWRuC4ZqRg="
```

elicoidal commented:
thanks @pedrobaeza
